stdlib/public/Concurrency/Task.cpp

```diff
@@ -1267,8 +1267,10 @@ static void swift_task_asyncMainDrainQueueImpl() {
 #if defined(_WIN32)
   static void(FAR *pfndispatch_main)(void) = NULL;
 
-  if (pfndispatch_main)
-    return pfndispatch_main();
+  if (pfndispatch_main) {
+    pfndispatch_main();
+    exit(0);
```

Comment:
... 🤦‍♂️ why did I not think of that?

Reply:
Actually, please change the exit code to something `[^0-2]`.  This is a failure case.

Reply:
If we want to assert this function can never return, but cannot do so with `_Noreturn` (and I couldn't figure out a way to make the compiler happy with that decoration on the function pointer above), we should use `swift_unreachable()`.